packages/react/src/hooks/useProvidedRefOrCreate.ts

```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+
+/**
+ * There are some situations where we only want to create a new ref if one is not provided to a component
+ * or hook as a prop. However, due to the `rules-of-hooks`, we cannot conditionally make a call to `React.useRef`
+ * only in the situations where the ref is not provided as a prop.
+ * This hook aims to encapsulate that logic, so the consumer doesn't need to be concerned with violating `rules-of-hooks`.
+ * @param providedRef The ref to use - if undefined, will use the ref from a call to React.useRef
+ * @type TRef The type of the RefObject which should be created.
+ */
+export function useProvidedRefOrCreate<TRef>(providedRef?: React.RefObject<TRef>): React.RefObject<TRef> {
```

Comment:
I debated pulling this in for ActionMenu too, but decided against it because in most cases you'll almost always need to provide the ref upfront, so there's no need to create one. Looking at where you're using it, perhaps that's worth considering here too? The less custom code we need to maintain the better IMO.

Reply:
Yup, that makes sense. I removed it, doesn't seem like we need it at all.